### PR TITLE
add a TODO

### PR DESCRIPTION
**Pull request by lightningbolts**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -225,6 +225,10 @@ function basic_simplify_expr(expr) {
   }
 }*/
 
+function simplify(expr) {
```
Comment (chengzh2008) [inline script.js:228]: what's this? Is it intentional?

File: script.js [hunk 2]
```diff
@@ -290,23 +297,26 @@ function derive_power(expr, variable) {
     return part2;
   }
   if (is_number(f) && is_number(g) !== true) {
-    const part1 = make_product(
+    /*const part1 = make_product(
```
Comment (chengzh2008) [inline script.js:300]: if it is not needed, you can simply remove it.

File: script.js [hunk 3]
```diff
@@ -290,23 +297,26 @@ function derive_power(expr, variable) {
...
-    const part2 = make_product(part1, gp);
+    );*/
+    const part2 = make_product(make_product(make_log(f), expr), gp);
     return part2;
   }
   if (is_number(f) !== true && is_number(g) !== true) {
     console.log("HEEERE")
     const part3 = make_product(make_log(f), g)
-    console.log(part3, "PART3")
+    console.log(display_expr(part3), "PART3")
```
Comment (chengzh2008) [inline script.js:310]: remove debug code once it is done.

File: script.js [hunk 4]
```diff
@@ -290,23 +297,26 @@ function derive_power(expr, variable) {
...
     const part2 = make_product(expr, derive(part3, "x"))
-    console.log(part2, "PART2")
+    console.log(display_expr(part2), "PART2")
     return part2
   }
 }
 
+function toView(value) {
```
Comment (chengzh2008) [inline script.js:317]: Is this being used?

**Comment by chengzh2008** [comment]
Looks good overall. Ship it!

File: script.js (hunk 1 above)
Comment (lightningbolts) [reply script.js:228]: yes

File: script.js (hunk 2 above)
Comment (lightningbolts) [reply script.js:300]: ok

File: script.js (hunk 3 above)
Comment (lightningbolts) [reply script.js:310]: fine

File: script.js (hunk 4 above)
Comment (lightningbolts) [reply script.js:317]: no

**Pull request merged**